File: Assignment 4/C/03. implement single linked list.c
#include <stdio.h>
#include <stdlib.h>

typedef struct node{
    int val;
    struct node* next;
}node;

int main(){
    node* head = (node*)malloc(sizeof(node));
    node* snd = (node*)malloc(sizeof(node));
    node* trd = (node*)malloc(sizeof(node));

    head->val = 1;
    head->next = snd;
    snd->val = 2;
    snd->next = trd;
    trd->val = 3;
    trd->next = NULL;

    node* ptr = head;
    while(ptr){
        printf("%d --> ",ptr->val);
        ptr = ptr->next;
    }
    printf("NULL\n");

    return 0;
}


//OUTPUT:

// 1 --> 2 --> 3 --> NULL

File: Assignment 4/Python/03. implement single linked list .py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

head = Node(1)
snd = Node(2)
trd = Node(3)

head.next = snd
snd.next = trd

ptr = head
while ptr:
    print(f"{ptr.val} --> ", end="")
    ptr = ptr.next
print("NULL")


# OUTPUT:
# 1 --> 2 --> 3 --> NULL
